java/dataset/src/test/java/org/apache/arrow/dataset/file/TestFileSystemDataset.java

```diff
@@ -357,6 +364,41 @@ public void testBaseArrowIpcRead() throws Exception {
     AutoCloseables.close(factory);
   }
 
+  @Test
+  public void testBaseOrcRead() throws Exception {
+    String dataName = "test-orc";
+    String basePath = TMP.getRoot().getAbsolutePath();
+
+    TypeDescription orcSchema = TypeDescription.fromString("struct<ints:int>");
+    Writer writer = OrcFile.createWriter(new Path(basePath, dataName),
+            OrcFile.writerOptions(new Configuration()).setSchema(orcSchema));
+    VectorizedRowBatch batch = orcSchema.createRowBatch();
+    LongColumnVector longColumnVector = (LongColumnVector) batch.cols[0];
+    longColumnVector.vector[0] = Integer.MIN_VALUE;
+    longColumnVector.vector[1] = Integer.MAX_VALUE;
+    batch.size = 2;
+    writer.addRowBatch(batch);
+    writer.close();
+
+    String orcDatasetUri = new File(basePath, dataName).toURI().toString();
+    FileSystemDatasetFactory factory = new FileSystemDatasetFactory(rootAllocator(), NativeMemoryPool.getDefault(),
```

Comment:
LGTM, only have this comment:

1. There is a [Jira ticket ](https://issues.apache.org/jira/browse/ARROW-17508)in case to use NativeMemoryPool.createListenable for big size data. Do you know if there are some limitation/restriction for big ORC files also?

Reply:
I don't know any things related to NativeMemoryPool.createListenable. It looks like common & separate Java Dataset API issue not introduced by this PR

Reply:
That ticket is entirely unrelated, I agree. (I added some comments on Jira.)

Reply:
Thanks for the clarification